pkg/actions/nodegroup/upgrade.go

```diff
@@ -46,19 +50,34 @@ func (m *Manager) upgradeAndWait(nodeGroupName, version, launchTemplateVersion s
 		Version:       &version,
 	}
 
+	describeNodegroupOutput, err := m.ctl.Provider.EKS().DescribeNodegroup(&eks.DescribeNodegroupInput{
+		ClusterName:   &m.cfg.Metadata.Name,
+		NodegroupName: &nodeGroupName,
+	})
+
+	if err != nil {
+		return err
+	}
+
 	if launchTemplateVersion != "" {
-		describeNodegroupOutput, err := m.ctl.Provider.EKS().DescribeNodegroup(&eks.DescribeNodegroupInput{
-			ClusterName:   &m.cfg.Metadata.Name,
-			NodegroupName: &nodeGroupName,
-		})
-		if err != nil {
-			return err
+		lt := describeNodegroupOutput.Nodegroup.LaunchTemplate
+		if lt == nil || (lt.Id == nil && lt.Name == nil) {
+			return errors.New("cannot update launch template version because the nodegroup is not configured to use one")
 		}
-
 		input.LaunchTemplate = &eks.LaunchTemplateSpecification{
 			Version: &launchTemplateVersion,
 			Id:      describeNodegroupOutput.Nodegroup.LaunchTemplate.Id,
+			Name:    describeNodegroupOutput.Nodegroup.LaunchTemplate.Name,
+		}
+	}
+
+	if version == "" {
+		// Use the current Kubernetes version
+		version, err := semver.ParseTolerant(*describeNodegroupOutput.Nodegroup.Version)
```

Comment:
So in this case the version isn't changed? Should/can the behavior be the same as [`create nodegroup`](https://github.com/weaveworks/eksctl/blob/7d7a8150bed6ccb6f58f82f634cbf05c3d479b57/pkg/actions/nodegroup/create.go#L242)?

Reply:
yes the version doesn't get changed. I just copied this from how it used to be to preserve the UX. Whats the behaviour in `create nodegroup`?

Reply:
If the behavior before was to not update anything when `--version` is left out then it should stay that way I suppose.

I think the default with `create nodegroup` is to use the control plane version, so here I suppose the version would be updated to the control plane version? Like I imagine I should be able to write `upgrade nodegroup --cluster xx --name xx-ng` and it would pick the right version.

Reply:
Perhaps a good change we could make as part of `v1.0`? wdyt?